AxonDeepSeg/visualization/simulate_axons.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+from scipy.misc import imrotate
+import imageio
+import AxonDeepSeg.ads_utils
+
+class SimulateAxons:
+
+    def __init__(self, image_dims=[1000,1000]):
+        self.width = image_dims[0]
+        self.height = image_dims[1]
+        self.origin = [self.width/2, self.height/2]
+
+        # Initialize image
+        self.image = np.zeros([self.width, self.height], dtype=np.uint8)
+
+    def generate_axon(self, axon_radius, center=None, gratio = 0.7, axon_angle = 0, plane_angle = 0, eccentricity = 0):
+        if center is None:
+            center=self.origin
+
+        axon_major_axis_radius = axon_radius/np.cos(np.deg2rad(plane_angle))
+
+        if eccentricity != 0:
+            axon_minor_axis_radius = np.sqrt(axon_radius**2-(eccentricity*axon_radius)**2)
+        else:
+            axon_minor_axis_radius = axon_radius
+
+        x = np.arange(0, self.width)
+        y = np.arange(0, self.height)
+        axon = (x[np.newaxis,:]-center[0])**2/(axon_major_axis_radius**2) + (y[:,np.newaxis]-center[1])**2/(axon_minor_axis_radius**2) < 1
+        
+        myelin_outer = (x[np.newaxis,:]-center[0])**2/(axon_major_axis_radius/gratio)**2 + (y[:,np.newaxis]-center[1])**2/(axon_minor_axis_radius/gratio)**2 < 1
```

Comment:
@mathieuboudreau I might be wrong, but it seems to me that the way the ellipse equation for the outer myelin is written, implies that each of the two axis is proportional to the axon_axis, using the same proportionality scaling (gratio). This implies that for a non-null eccentricity, the myelin will be thicker in in the major axis, and thinner in the minor axis. Realistically (I think), the myelin sheath has the same density regardless the eccentricity. This is a different story with the plane_angle parameter though. I made a [sketch](https://photos.app.goo.gl/5GdZTjrXC9hqv6ZCA) to describe the different scenario.

Reply:
Yeah, I was still parsing in my mind what cases should considered, so your input is very valuable. Right now, I think varying both the `plane_angle` or the `eccentricity` are simulating the same thing; a non-perpendicular plane slice of the cylinders. The difference between the two is that increasing the `plane_angle` will make the ellipsoidal slice of the axon & myelin "longer" in terms of pixels (at is would actually be irl), whereas controlling the `eccentricity` parameter allows you to fix the major axis (this could be useful in terms of simulations to avoid too much overlap when planning multiple axons in an image, since varying `plane_angle` makes their elliptical length unknown from a user's standpoint).

Neither of these can be used to simulate a non-circular cylindrical cut at the moment, i.e. a plane through an an [elliptic cylinder](http://mathworld.wolfram.com/EllipticCylinder.html) or axon, which I think is what your scenario 2 (middle) in your image is displaying, correct? At the moment, this branch is only considering circular cylinders as axons, and the ellipses come from a plane at an angle from it.

Reply:
Case #3 that you showed (with "thicker" myelin at the major axis, and "thiner" at the minor) is what would really happen if you cut an axon at a non-perpendicular angle, I believe.

Reply:
I think the major issue at the moment, is I should have clarified that `eccentricity` is not the eccentricity of the **axon**, but of the **ellipse**.